### ISSUE TITLE
CLI-842: Using domain-based multisite - `acli pull:db` creates a new database instead of writing over main

### ISSUE THREAD

**Issue by woodland-solutions-group** [acquia/cli#1152]
Noticed this issue because Cloud IDE recently updated ACLI to 2.20 in an automatic update and our local dev process broke. I'm not sure what version of ACLI we were on before the updated. More on versions later.

Our process has been to run `acli pull:db` to change the site we're working on. With 2.20 this command creates a new database instead of replacing content in the database in use. For example, Cloud IDE uses a database called _drupal_. Instead of replacing content in _drupal_ if we run `acli pull:db` to get content for a site called *example_1* then there will be a new database alongside _drupal_ called *example_1*. If we then run `acli pull:db` to get another database called *example_2* then we will have 3 databases alongside each other called _drupal_, *example_1*, and *example_2*.  The problem is that our site continues to point to _drupal_ because it's domain-based.

Bringing this up because we've been using ACLI for over a year with it working well for database switching on our site. We're using a copy of ACLI 1.19 to achieve our old, expected results and it's working as expected. We selected 1.19 because we raised the issue with Acquia that prompted the 1.18.1 release, so we knew 1.19 would work as expected.

**Comment by anavarre** [comment]
@woodland-solutions-group thanks for the report. We also got similar reports yesterday. This is indeed happening as of 2.2.0 and because of https://github.com/acquia/cli/pull/1121 - For now your solution is to downgrade to 2.1.0 or below.

> With 2.20 this command creates a new database instead of replacing content in the database in use.

The reason why this is happening is we have a goal to better support Drupal multisite in Coud IDE and the intent is to no longer rely on the `drupal` database (which you are forced to overwrite anytime you wish to pull a site into the IDE) but to have the same database(s) created within as the one(s) you have on the Cloud Platform.

> The problem is that our site continues to point to drupal because it's domain-based.

Now that you understand the Acquia CLI intent, let's talk about Cloud IDE. The work to be done here is to support sub-domains so you can serve multiple URLs instead of only the default one you have access to ATM. This is what the team is currently working on.

@danepowell FYI

**Comment by anavarre** [comment]
https://github.com/acquia/cli/releases/tag/2.3.1 was released yesterday and should fix your issue since it brings back the regular behavior for now.

**Comment by woodland-solutions-group** [comment]
Updated ACLI, cleared caches, and tried to import a DB again. Same results as listed in the issue. I looked at the activity in #1159 and wonder if part of the problem is that have a domain-based multisite. Maybe this is a known configuration issue.

We're back to our old workflow using v1.30.1 a the moment. Side-note: I tried v2.1.0 but ran into PHP 8 problems. We're planning to migrate Dev to PHP 8 Friday and I'll try v2.1.0 again.
